### PR TITLE
add tests to latex definitions files

### PR DESCRIPTION
**Pull request by kaligrafy**
verify consistence between all languages and verify possible duplicates

### CODE REVIEW

**Comment by kaligrafy** [comment]
needs rebasing on https://github.com/chairemobilite/transition/pull/1158

File: packages/transition-backend/file/__tests__/definitions.test.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import fs from 'fs';
+
+// Language configuration - TODO: this should be moved to a separate config file
+const languages = [
+    'en',
```
Comment (greenscientist) [inline packages/transition-backend/file/__tests__/definitions.test.ts:5]: Add a comment that say that the first one is the reference language

File: packages/transition-backend/file/__tests__/definitions.test.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            );
+
+            if (missingLabels.length > 0) {
+                throw new Error(
+                    `Missing labels in ${lang}:\n${missingLabels.join('\n')}`
+                );
+            }
+        });
+
+        test(`all ${lang} labels exist in English file`, () => {
```
Comment (greenscientist) [inline packages/transition-backend/file/__tests__/definitions.test.ts:50]: Should name the test "labels exist in reference language file"

File: packages/transition-backend/file/__tests__/definitions.test.ts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            );
+
+            if (extraLabels.length > 0) {
+                throw new Error(
+                    `Extra labels in ${lang} not present in English:\n${extraLabels.join('\n')}`
+                );
+            }
+        });
+
+        test(`${lang} has same number of labels as English`, () => {
```
Comment (greenscientist) [inline packages/transition-backend/file/__tests__/definitions.test.ts:62]: Same

File: packages/transition-backend/file/__tests__/definitions.test.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            expect(labelSets[lang].size).toBe(labelSets[referenceLanguage].size);
+        });
+
+        test(`labels appear in same order as English in ${lang} file`, () => {
+            const langLabels = [...labelSets[lang]];
+            const englishLabels = [...labelSets[referenceLanguage]];
```
Comment (greenscientist) [inline packages/transition-backend/file/__tests__/definitions.test.ts:68]: Change englishLabels for referenceLanguageLabels

File: packages/transition-backend/file/__tests__/definitions.test.ts [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+// Language configuration - TODO: this should be moved to a separate config file
+const languages = [
+    'en',
+    'fr'
+];
+
+describe('LaTeX Label Consistency Tests', () => {
+    // Helper function to extract labels from LaTeX content
+    function extractLabels(content) {
+        const labelRegex = /\\label{([^}]+)}/g;
```
Comment (GabrielBruno24) [inline packages/transition-backend/file/__tests__/definitions.test.ts:12]: Give a short description of what the regex does (specifically the part after 'label')

File: packages/transition-backend/file/__tests__/definitions.test.ts (hunk 1 above)
Comment (kaligrafy) [reply packages/transition-backend/file/__tests__/definitions.test.ts]: added

File: packages/transition-backend/file/__tests__/definitions.test.ts (hunk 5 above)
Comment (kaligrafy) [reply packages/transition-backend/file/__tests__/definitions.test.ts:12]: added

**Review by GabrielBruno24 — APPROVED** [review]
Yannick's comments should be adressed, but I approve after that.

**Review by greenscientist — APPROVED** [review]

**Pull request merged**